test/HyVM/Opcodes.t.sol

```diff
@@ -1206,4 +1206,14 @@ contract OpcodesTest is Test {
         (bool success, ) = hyvm.call(bytecode);
         assertTrue(success, "expectRevert: call did not revert");
     }
+
+    function testPCRetrievalByOverFlow() public {
+        // bytecode generated using: easm test/opcodes/pc-retrieval-by-overflow
+        // simple mstore that will store to far in memory
+        vm.expectRevert(bytes(""));
+        (bool success,) = hyvm.delegatecall(
+            hex"7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff51"
+        );
+        assertTrue(success, "expectRevert: call did not revert");
```

Comment:
I believe this wont revert, given that it will just go out-of-gas ?

Reply:
With `out of gas` error the `delegatecall` will return false. Hence, the cheatcode `expectRevert` can be used.
`Out of gas` error are equivalent to a revert as the whole transaction would revert except when using low level call or if the call is inside a `try/catch`

Reply:
Yup, but you'll have to pass it a lesser gas limit than your outer gas limit (otherwise, you wont have any gas left to check for success ^^) .... I suppose that's the case here ?

Reply:
yup, due to https://eips.ethereum.org/EIPS/eip-150 which is enough

Reply:
With the [63/64 rule](https://www.rareskills.io/post/eip-150-and-the-63-64-rule-for-gas) there is not need to take this into account
You will always have enough gas to handle the return of the subcall